src/vdom.rs

```diff
@@ -288,9 +328,46 @@ impl<Ms, Mdl, ElC: View<Ms> + 'static, GMs: 'static> App<Ms, Mdl, ElC, GMs> {
         since = "0.4.2",
         note = "Please use `AppBuilder.build_and_start` instead"
     )]
-    pub fn run(self) -> Self {
+    pub fn run(mut self) -> Self {
+        let AppRunCfg {
+            mount_type,
+            into_after_mount,
+            ..
+        } = self
+            .run_cfg
+            .take()
+            .expect("run_cfg should be set in App::new which is called from AppBuilder");
+
         // Bootstrap the virtual DOM.
-        self.data.main_el_vdom.replace(Some(self.bootstrap_vdom()));
+        self.data
+            .main_el_vdom
+            .replace(Some(self.bootstrap_vdom(mount_type)));
+
+        // Can this simply be `self.update`?
+        let mut orders = OrdersContainer::new(self.clone());
+        let builder::AfterMount {
+            model,
+            url_handling,
+        } = into_after_mount.into_after_mount(routing::initial_url(), &mut orders);
+
+        self.data.model.replace(Some(model));
+
+        // TODO: Does this go before or after setting up the routes listener?
+        match url_handling {
+            UrlHandling::PassToRoutes => {
+                let url = routing::initial_url();
+                if let Some(r) = self.data.routes.borrow().as_ref() {
+                    if let Some(u) = r(url) {
```

Comment:
I don't know if it's old copy/pasted code or a new one, but those one-letter variables are hard to read.

Reply:
Noted. They were copy and pasted. I'll fix them soon.